crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -195,13 +204,16 @@ fn is_callee_a_promise(callee: &AnyJsExpression, model: &SemanticModel) -> bool
             let Some(any_js_binding_decl) = binding.tree().declaration() else {
                 return false;
             };
-
-            let AnyJsBindingDeclaration::JsFunctionDeclaration(func_decl) = any_js_binding_decl
-            else {
-                return false;
-            };
-
-            return is_function_a_promise(&func_decl);
+            match any_js_binding_decl {
+                AnyJsBindingDeclaration::JsFunctionDeclaration(func_decl) => {
+                    return is_function_a_promise(&func_decl);
+                }
+                AnyJsBindingDeclaration::JsVariableDeclarator(js_var_decl) => {
+                    return is_variable_initializer_a_promise(&js_var_decl)
+                        || is_variable_annotation_a_promise(&js_var_decl);
+                }
+                _ => {}
+            }
```

Comment:
Nit: if you remove the `false` at the end, you can just use this entire `match` as an expression that defines the return value, so you don't need the repeated `return` statements. This is generally considered a bit more idiomatic in Rust:

```suggestion
            match any_js_binding_decl {
                AnyJsBindingDeclaration::JsFunctionDeclaration(func_decl) => {
                    is_function_a_promise(&func_decl)
                }
                AnyJsBindingDeclaration::JsVariableDeclarator(js_var_decl) => {
                    is_variable_initializer_a_promise(&js_var_decl)
                        || is_variable_annotation_a_promise(&js_var_decl)
                }
                _ => false
            }
```

Reply:
Thanks for pointing that out! 
I've updated the code to remove the repeated return statements. It looks cleaner and more idiomatic now.